src/components/CreateCardPage.js

```diff
@@ -1,139 +1,170 @@
+import PropTypes from 'prop-types'
 import React, { useState } from 'react'
-import styled from 'styled-components/macro'
-import Img from '../components/cards/Img'
-import Page from '../common/Page'
 import { Redirect } from 'react-router-dom'
+import styled from 'styled-components/macro'
 import { ImageAdd } from 'styled-icons/boxicons-regular/ImageAdd'
-import logo from '../data/petshare.png'
-import axios from 'axios'
-import petIcon from '../data/paw-solid.svg'
-import createCard from '../data/CreateCard.png'
-
-const CLOUDNAME = process.env.REACT_APP_CLOUDINARY_CLOUDNAME
-const PRESET = process.env.REACT_APP_CLOUDINARY_PRESET
-
-
+import Page from '../common/Page'
+import createCard from '../images/CreateCard.png'
+import petIcon from '../images/paw-solid.svg'
+import { uploadPicture } from '../utils/cardServices'
 
-export default function CreateCardPage({editCardData, onSubmit}) {
+CreateCardPage.propTypes = {
+  editcardData: PropTypes.object,
+  onSubmit: PropTypes.func.isRequired
+}
 
-  const [title, setTitle] = useState(editCardData.title ? editCardData.title : '' )
-  const [description, setDescription] = useState(editCardData.description ? editCardData.description : '')
-  const [email, setEmail] = useState(editCardData.email ? editCardData.email : '')
+export default function CreateCardPage({ editCardData, onSubmit }) {
+  const [title, setTitle] = useState(
+    editCardData.title ? editCardData.title : ''
+  )
+  const [description, setDescription] = useState(
+    editCardData.description ? editCardData.description : ''
+  )
+  const [email, setEmail] = useState(
+    editCardData.email ? editCardData.email : ''
+  )
   const [tags, setTags] = useState(editCardData.tags ? editCardData.tags : '')
-  const [picture, setPicture] = useState(editCardData.picture ? editCardData.picture : '')
+  const [picture, setPicture] = useState(
+    editCardData.picture ? editCardData.picture : ''
+  )
   const [isCreated, setIsCreated] = useState(false)
-  
+
   function handleSubmit(event) {
     event.preventDefault()
     const form = event.target
     const formData = new FormData(form)
     const data = Object.fromEntries(formData)
-        data.tags = {
-          type: data.type,
-          size: data.size,
-          availability: data.availability
-        }
+    data.tags = {
+      type: data.type,
+      size: data.size,
+      availability: data.availability
+    }
     data.picture = picture
     const newEditCardData = {
       ...editCardData,
-          title, description, email, tags, picture
+      title,
+      description,
+      email,
+      tags,
+      picture
     }
-      
+
     if (editCardData.id) {
       onSubmit(editCardData.id, newEditCardData)
       setIsCreated(true)
     } else {
       onSubmit(data)
       setIsCreated(true)
```

Comment:
In beiden Fällen setzt du 'setIsCreated(true)' also kannst du das auch nach dem if-else nur einmal setzen. Und dann hast du nur jeweils eine Sache bei den if-else, dann sieht ein ternary operator besser aus.